src/views/Home.jsx

```diff
@@ -1,12 +1,58 @@
 import './Home.css';
+import { useCallback, useState } from 'react';
+import { checkListToken } from '../api/firebase';
+import { useNavigate } from 'react-router-dom';
+
+export function Home({ handleNewToken, setListToken, listToken }) {
+	const [token, setToken] = useState('');
+	const [listNotFound, setlistNotFound] = useState('');
+	const redirect = useNavigate();
+
+	if (listToken) {
+		redirect('/list');
+	}
+
+	const handleClick = useCallback(() => {
+		handleNewToken();
+	}, [handleNewToken]);
+
+	const handleFormChange = (e) => {
+		const value = e.target.value;
+		setToken(value);
+	};
+
+	const handleTokenSubmit = async (e) => {
+		e.preventDefault();
+
+		const checkTokenExists = await checkListToken(token);
+
+		if (checkTokenExists) {
+			setListToken(token);
+			redirect('/list');
+		} else {
+			setlistNotFound('Could not find this list');
+		}
+	};
 
-export function Home({ handleNewToken }) {
 	return (
 		<div className="Home">
 			<p>
 				Hello from the home (<code>/</code>) page.
 			</p>
-			<button onClick={handleNewToken}>Create a new list!</button>
+			<button onClick={handleClick}>Create a new list!</button>
+			<p> Want to join an existing list? </p>
+			<form onSubmit={handleTokenSubmit}>
+				<label htmlFor="token"> Enter Token:</label>
```

Comment:
I know it's a token, but is this accessible for someone who isn't computer literate?  Just a thought, you probably already had it too.

Reply:
Good point! Updated with aria-describedby description as per @mxmason